src/Lecture1.hs

```diff
@@ -108,7 +119,21 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Int
+strSum str = calcIt 0 str
+    where calcIt summ [] = summ
+          calcIt summ str1 = calcIt (summ + getInt (takeWhile charu (dropWhile space str1))) (dropWhile charu (dropWhile space str1))
```

Comment:
You can use standard functions to simplify the implementation of `strSum`:

* `words`: to split a string into a list of space-separated words
* `read`: to parse an `Int` from `String`
* `sum`: to find a sum of all numbers in the list

Reply:
Yupp... I have to start forcing myself to use more prelude functions 😅.